### ISSUE TITLE
Support for official Google Play Games apk (at least minimal)

### ISSUE THREAD

**Issue by ale5000-git** [microg/android_packages_apps_GmsCore#163]
It would be nice to have this :)

It currently just say that there is a problem in the Google Play Services.
Here is the logcat to get some hint: https://www.zerobin.net/?0e026f4c2fa6a878#1BUMnxcFvzeF5Li3X1wi/ISY4K2vBWnOoCRwHgMhkVE=

**Comment by Technowix** [comment]
Any news on this ? :)

**Comment by Technowix** [comment]
How can we pay the bounty ? :p

**Comment by ale5000-git** [comment]
@Technowix: I get a notification of your comment but I don't see it.
The bounty can be paid / collected here: https://www.bountysource.com/

**Comment by Technowix** [comment]
@ale5000-git Yeah, I was thinking someone asked 5$ to get this working, and deleted my post after I saw I where wrong xD, I will add 5$ more by few days I think :) !

**Comment by ShapeShifter499** [comment]
Is this a issue for some people?  The official Google Play apk is working for me with only microG services installed.  

My system
Moto X Pure arm64
Cyanogenmod 13.1 Nightly 11-21-2016 Build
Xposed v0.87
Xposed Signature Module installed

**Comment by ShapeShifter499** [comment]
I should add I have the play store installed to /system/priv-app  and I gave it permissions for storage access and nothing else.

**Comment by ShapeShifter499** [comment]
I deleted my previous comments, I misunderstood the issue here, sorry.

**Comment by ale5000-git** [comment]
Logcat from CyanogenMod 14.1 => [Google Play Games problem.txt](https://github.com/microg/android_packages_apps_GmsCore/files/638091/Google.Play.Games.problem.txt)

**Comment by ale5000-git** [comment]
Has anyone tested microG with [this](http://repo.xposed.info/module/com.fifsource.android.noplaygames)?

It doesn't fix this issues but it is good workaround in the mean time.

**Comment by Technowix** [comment]
It simply disable the gplay hint right ? My game need to log in to fetch my account :(

**Comment by jurek23452** [comment]
http://pastebin.com/wBxKpMZ3 Logcat Latest build from source code android 6.0 cyanogenmod

**Comment by ale5000-git** [comment]
@Technowix: I think yes but I haven't tried.
Could you mention a game that use it? Just for reference.

**Comment by Technowix** [comment]
@ale5000-git I'm targeting _Dynamix_ to work, but as I said they rely on the google play framework to store/manage paid account (that I have for unlocking more songs to play)

**Comment by Ningyoplug** [comment]
Love Live! School Idol Festival also uses Google Play Games to store account info. I'm trying to access my account that used to be on another device but can't because Google Play Games won't work. :c

**Comment by licaon-kter** [comment]
[NEO Scavenger](https://play.google.com/store/apps/details?id=com.bluebottlegames.neoscavengermobile) yields this crash: https://gist.github.com/licaon-kter/920c77d2cec698cd993665c07b150286

**Comment by MOVZX** [comment]
Clash of Clans requires Google Play Games API.

**Comment by licaon-kter** [comment]
Regarding NEO Scavenger:
* on one device with microG - black screen
* on another device with microG - works ok
* on one device with Google Service (disabled/enabled/updated/etc) - black screen

:-| strange

**Comment by ShapeShifter499** [comment]
Bounty is now $150

**Comment by juskdtlm** [comment]
Same problem

**Comment by AndnixSH** [comment]
Anyone will care about this?
Many users using unsigned apks on virtualxposed app but MicroG doesn't work with Google Play Games. Please do something

**Comment by Hae14** [comment]
aww.. 'til now.. No One?

**Comment by ale5000-git** [comment]
@nm111 has worked on it here: https://github.com/microg/android_external_GmsApi/pull/12
But I'm not sure what is the current state.

**Comment by skid9000** [comment]
I don't know if the pr is in the last release but Azur Lane and Love Live SIFAS crash right after the loading screen :/

**Comment by skid9000** [comment]
Wait what the... ?!

After flashing some random magisk modules, i successfully launched Love Live SIFAS, **without passing safety-net**

**Comment by lyndmi** [comment]
Nice to see someone already working on Play Games

**Comment by skid9000** [comment]
Here is the logcat of AzurLane if someone want to investigate...

[azur.txt](https://github.com/microg/android_packages_apps_GmsCore/files/4007609/azur.txt)

**Comment by AndnixSH** [comment]
Workaround is to use VM for now using VMOS. Official Google services working perfectly on it and it can be rooted
My dream became true lol
https://www.vmos.com/

**Comment by skid9000** [comment]
OOF https://github.com/anbox/anbox/issues/1108

**Comment by AndnixSH** [comment]
Who cares? VMOS works way better than VirtualXposed and can install unsigned APKs with working Google services. No one care about Google Play Games anyway so this is only option

I don't know much about the GPL law either

**Comment by lyndmi** [comment]
Wow didnt know you can run vm on phone. Tested, it's working. Google play games working perfectly yaay!

**Comment by nm111** [comment]
I honestly don't see much point in choosing uG and then installing GMS again but just in a container. Isn't the point of all this to remove Play Services entirely?
GPL is essentially if you distribute (modified) copies or substantial portions of the program, these copies must retain the license and source code for them must be also available. So it is clear that VMOS is violating this. There exists https://gpl-violations.org/, there is some information there.

**Comment by Miaourt** [comment]
@nm111 sadly as many of the people there know, it's almost impossible to use most now common android apps without Google's services.
So being able to use them with microg is fine, being able to use them in a isolated environment might be even easier and better privacy wise, since they are isolated ?

**Comment by lyndmi** [comment]
@Miaourt yes much easier especially for running multiple accounts on non root devices. I was using Virtualxposed too that has microG installer but play games don't work. Parallel space support google services but outdated.
VMOS is actually more compatibility than those parallel apps

**Comment by ale5000-git** [comment]
We should be talking about re-implementing apis needed for Play Games here not about a thing not related.

**Comment by lyndmi** [comment]
I don't think more devs are willing to waste time implementing it than a single user who tried. It's still a lack of interest :(

**Comment by BMD4E** [comment]
I bet this developer is too lazy, thats why no implement and many issues still open
No implement makes VX totally worthless
i will tell users to switch to VM OS

by the way tutorial how to install unsigned apk on non-root device brought me here ;)

**Comment by anonimou0** [comment]
Hey @BMD4E .

Don't be an ass hole calling others lazy and let us see your PRs.

**Comment by nift4** [comment]
Just for clarifictaion: The thing that user did [here](https://github.com/microg/android_external_GmsApi/pull/12) is only an definition on how games communicate with Google Play Games, not any work to get the Play Games app work.

**Comment by Edeae833** [comment]
But he deleted his Github account after that :(

**Comment by nift4** [comment]
because he didn't need it anymore

**Comment by DeathGod2k1** [comment]
160$ Bounty now

**Comment by Toshiba1127** [comment]
This project is dead already, no one care about bounty

**Comment by nift4** [comment]
nope, there are updates recently as you see. and this is just hard and not high priority. why don't you just add Play Games support?

**Comment by Toshiba1127** [comment]
I'm not a developer. You can't just tell somebody like that

**Comment by chris42** [comment]
You mean it is impolite to say? Like "This project is dead already, no one care about microG and bounty"?

It is how Open Source works. If someone has the time he can work on the topics. If your topic is not high priority you will need more people with more time. So find someone who is willing to program it for $160 or learn to do it yourself. If both is not an option you have to wait.

**Comment by VanHoevenTR** [comment]
Is bounty $210 or still $160? i'm confused

**Comment by MartinX3** [comment]
Please click on the bounty link in the first post for up to date results.
There must be an issue with the bounty bot writing the wrong title.

**Comment by VanHoevenTR** [comment]
Okay thank you

**Comment by ajddavid452** [comment]
wait I'm confused is this for the app or do the actual games themselves have issues?

**Comment by ale5000-git** [comment]
I think some Games require Google Play Games while on others it may be optional or not used at all.

**Comment by ajddavid452** [comment]
> I think some Games require Google Play Games while on others it may be optional or not used at all.

oh I see

**Comment by AtFreezingPoint** [comment]
Still nothing on this?

**Comment by manuth** [comment]
Just throwing in the fact, that "Ace Attorney Investigations: Miles Edgeworth" is affected by this as well.
If it is of any help, I could set up /e/OS on an Android device in order to collect logs.

Would that be of any help to some upcoming developer willing to implement this?

**Comment by manuth** [comment]
Alright - thank you so much for your rapid answer 😄

**Comment by licaon-kter** [comment]
@account183892 there were some game fixes in latest versions, did you update to 0.3.2 and tested?

**Comment by cetio** [comment]
Bounty still active?

**Comment by MartinX3** [comment]
> Bounty still active?

Please use a search engine.

**Comment by ale5000-git** [comment]
Probably this PR will help a lot: #2576
But I haven't tried it yet.

PS: Unfortunately bountysource is died and the money are probably lost.
